web/src/components/insight-page/insight-header/index.tsx

```diff
@@ -12,8 +12,7 @@ import { Autocomplete } from "@material-ui/lab";
 import { FC, memo, useState, useEffect } from "react";
 import { useAppDispatch, useAppSelector } from "~/hooks/redux";
 import { Application, selectAll, selectById } from "~/modules/applications";
-import { fetchDeploymentChangeFailureRate } from "~/modules/deployment-change-failure-rate";
-import { fetchDeploymentFrequency } from "~/modules/deployment-frequency";
```

Comment:
The header shouldn't be dependent on other components.

Reply:
I think it's ok for the header to depend on some modules because it's literally a module. wdyt?
BTW, I feel it's better to combine these modules into one.

Reply:
Because `fetchApplicationCount` and other API calls are being executed in the `insight-page`, so moving those API calls gives us unification.
And in my opinion, the child components like `insight-header` should avoid interacting directly with the API.
They should only care about their rendering and receive the needed data from their parent.

Reply:
Ah, I got your points and looks good.